File: src/main/java/ru/sfedu/teamselection/repository/TrackRepository.java
package ru.sfedu.teamselection.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.transaction.annotation.Transactional;
import ru.sfedu.teamselection.domain.Track;

@Transactional
public interface TrackRepository extends JpaRepository<Track, Long> {
//    List<Track> findAllByType(String type);
}

File: src/main/kotlin/ru/sfedu/teamselection/repository/TrackRepository.kt
package ru.sfedu.teamselection.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.transaction.annotation.Transactional
import ru.sfedu.teamselection.domain.Track

@Transactional
interface TrackRepository: JpaRepository<Track, Long>{
    fun findAllByType(type: String): List<Track>
}